daq/base_gateway.py

```diff
@@ -224,6 +229,49 @@ def terminate(self):
                 LOGGER.error('Gateway %s terminating dummy: %s', self.name, e)
                 LOGGER.exception(e)
 
+    def _get_scan_interface(self):
+        return self.host, self.host_intf
+
+    def discover_host(self, subnets: List[ip_network], mac: str, callback: Callable, vid=None) -> ip_address:
+        cmd = 'arp-scan --retry=2 --bandwidth=512K --interface=%s --destaddr=%s %s %s'
+        host, intf = self._get_scan_interface()
+        LOGGER.info('Starting host discovery for %s', mac)
+        def hangup_callback(log_fd, log_file):
```

Comment:
Do these functions (hangup_callback, process_line) need to be inlined with the main function (discover_host)?  I know it makes some things easier, but typically I find it harder to read (overall function is ~40 lines when it could be broken up into smaller subunits). Seems like hangup_callback doesn't have much borrowed from the discover_host context (just callback, which could be passed with a lambda).

Reply:
I can clean these up but is there any other concern with this way of discovering hosts? It seems to work very well (for ipv4 at least).

Reply:
Ok -- just making sure I understand what's going on.  The basics are that when a host times out waiting for host DHCP it will call this function with the mac address of the host, and then this calls arp-scan on that host which will send it an ARP to try and figure out it's IP address?   It's not, for example, trying to periodically scan all IP addresses in the subnet or something like that?

If so (hitting a single host), then I think it's perfectly fine.

Reply:
So yes, when a host times out when using external DHCP, it'll try to figure out host's IP using arp-scan. Other dhcp times out aren't affected.